geoips/create_plugin_registries.py

```diff
@@ -523,139 +523,141 @@ def add_yaml_plugin(filepath, relpath, package, plugins):
-                # available in the same plugin package as the current plugin,
-                # so pulling the family and docstring from the product defaults
-                # when creating the registry will not always work.
-                # If we are concerned about efficiency with opening product and
-                # product_defaults with every plugin access, we could do a
-                # second pass in the plugin registry creation to fill in the
-                # nulls (but we will not worry about that yet)
-                # if not docstring or not family:
-                #     # if the yaml_subplg doesn't include a docstring, grab its
-                #     # product_defaults docstring
-                #     if (
-                #         "product_defaults" not in plugins
-                #         or pd not in plugins["product_defaults"]
-                #     ):
-                #         LOG.error(
-                #             f"""Product defaults '{pd}' does not exist.
-                #                   Using 'undefined' docstring.
-                #                   Need to figure out how to pull product defaults
-                #                   from a different plugin package at some point."""
-                #         )
-                #     else:
-                #         if not docstring:
-                #             docstring = plugins["product_defaults"][pd]["docstring"]
-                #         if not family:
-                #             family = plugins["product_defaults"][pd]["family"]
-                plugins[interface_name][subplg_source][subplg_product] = {
-                    "docstring": format_docstring(docstring),
-                    "family": family,
-                    "interface": interface_module.name,
-                    "package": plugin["package"],
-                    "plugin_type": "yaml_based",
-                    "product_defaults": pd,
-                    "source_names": source_names,
-                    "relpath": plugin["relpath"],
-                }
-    else:
-        error_message += check_plugin_exists(
-            package, plugins, interface_name, plugin["name"], relpath
-        )
+    for plugin in yaml.load_all(open(filepath, mode="r"), Loader=yaml.SafeLoader):
+        plugin["relpath"] = relpath
+        plugin["package"] = package
+
+        try:
+            interface_name = plugin["interface"]
+        except KeyError:
+            raise PluginRegistryError(
+                f"""No 'interface' level in '{filepath}'.
+                    Ensure all required metadata is included."""
+            )
+        interface_module = getattr(geoips.interfaces, f"{interface_name}")
+
+        if interface_name not in plugins.keys():
+            plugins[interface_name] = {}
+
+        error_message = ""
+        # If the current family is "list", make sure we loop through the list,
+        # expanding out each individual product found within the list.
```

Comment:
Is everything below this code block unchanged?  Just indented under the yaml.load_all ?

Reply:
Yes, nothing else changed.